main/src/ca/uwaterloo/flix/language/phase/extra/CodeHinter.scala

```diff
@@ -15,350 +16,170 @@
 
-    case Expr.Tuple(exps, _, _, _) =>
-      visitExps(exps)
-
-    case Expr.RecordEmpty(_, _) => Nil
-
-    case Expr.RecordSelect(exp, _, _, _, _) =>
-      visitExp(exp)
-
-    case Expr.RecordExtend(_, exp1, exp2, _, _, _) =>
-      visitExp(exp2) ++ visitExp(exp1)
-
-    case Expr.RecordRestrict(_, exp, _, _, _) =>
-      visitExp(exp)
-
-    case Expr.ArrayLit(exps, exp, _, _, _) =>
-      visitExps(exps) ++ visitExp(exp)
-
-    case Expr.ArrayNew(exp1, exp2, exp3, _, _, _) =>
-      visitExp(exp1) ++ visitExp(exp2) ++ visitExp(exp3)
-
-    case Expr.ArrayLoad(exp1, exp2, _, _, _) =>
-      visitExp(exp1) ++ visitExp(exp2)
-
-    case Expr.ArrayStore(exp1, exp2, exp3, _, _) =>
-      visitExp(exp1) ++ visitExp(exp2) ++ visitExp(exp3)
-
-    case Expr.ArrayLength(exp, _, _) =>
-      visitExp(exp)
-
-    case Expr.StructNew(_, fields, region, _, _, _) =>
-      fields.map { case (_, v) => v }.flatMap(visitExp) ++ visitExp(region)
-
-    case Expr.StructGet(exp, _, _, _, _) =>
-      visitExp(exp)
-
-    case Expr.StructPut(exp1, _, exp2, _, _, _) =>
-      visitExp(exp1) ++ visitExp(exp2)
-
-    case Expr.VectorLit(exps, _, _, _) =>
-      visitExps(exps)
-
-    case Expr.VectorLoad(exp1, exp2, _, _, _) =>
-      visitExp(exp1) ++ visitExp(exp2)
-
-    case Expr.VectorLength(exp, _) =>
-      visitExp(exp)
-
-    case Expr.Ascribe(exp, _, _, _) =>
-      visitExp(exp)
-
-    case Expr.InstanceOf(exp, _, _) =>
-      visitExp(exp)
-
-    case Expr.CheckedCast(_, exp, _, _, _) =>
-      visitExp(exp)
-
-    case Expr.UncheckedCast(exp, _, _, _, _, _) =>
-      visitExp(exp)
-
-    case Expr.Without(exp, _, _, _, _) =>
-      visitExp(exp)
+  /**
+    * Returns a [[CodeHintCandidates]] for the Flix project.
+    *
+    * @param root The root AST node of the Flix project
+    * @return     A [[CodeHintCandidates]] for the Flix project.
+    */
+  private def getCandidates(implicit root: Root): CodeHintCandidates = {
+    var traitUses: List[TraitSymUse] = Nil
+    var defUses: List[DefSymUse] = Nil
+    var enumUses: List[(Symbol.EnumSym, SourceLocation)] = Nil
+    var defCalls: List[(Symbol.DefnSym, List[Expr])] = Nil
 
-    case Expr.TryCatch(exp, rules, _, _, _) =>
-      visitExp(exp) ++ rules.flatMap {
-        case CatchRule(_, _, exp) => visitExp(exp)
+    object UseConsumer extends Consumer {
+      override def consumeTraitSymUse(symUse: TraitSymUse): Unit = traitUses = symUse :: traitUses
+      override def consumeTraitConstraintHead(tcHead: TraitConstraint.Head): Unit = traitUses = TraitSymUse(tcHead.sym, tcHead.loc) :: traitUses
+      override def consumeDefSymUse(sym: DefSymUse): Unit = defUses = sym :: defUses
+      override def consumeExpr(exp: Expr): Unit = exp match {
+        case TypedAst.Expr.ApplyDef(symUse, exps, _, _, _, _) => defCalls = (symUse.sym, exps) :: defCalls
```

Comment:
We don't use the term "call". We use "Apply" so call it applyDefs (this comment applies everywhere).